src/ripple/overlay/impl/ProtocolMessage.h

```diff
@@ -226,23 +263,35 @@ invokeProtocolMessage(Buffers const& buffers, Handler& handler)
     if (size == 0)
         return result;
 
-    auto header = detail::parseMessageHeader(buffers, size);
+    auto header = detail::parseMessageHeader(result.second, buffers, size);
 
     // If we can't parse the header then it may be that we don't have enough
-    // bytes yet, or because the message was cut off.
+    // bytes yet, or because the message was cut off (if error_code is success).
+    // Otherwise we failed to match the header's marker (error_code is set to
+    // no_message) or the compression algorithm is invalid (error_code is
+    // protocol_error) and signal an error.
     if (!header)
         return result;
 
     // We implement a maximum size for protocol messages. Sending a message
     // whose size exceeds this may result in the connection being dropped. A
     // larger message size may be supported in the future or negotiated as
     // part of a protocol upgrade.
-    if (header->payload_wire_size > megabytes(64))
+    if (header->payload_wire_size > megabytes(64) ||
+            header->uncompressed_size > megabytes(64))
```

Comment:
Nice catch on this one.

Also, nit: Clang format complains about this: no extra indent for second line of the condition.